todo_app/app.py

```diff
@@ -28,26 +36,103 @@ def create_app():
         'DELETE':todoCards.delete_one
     }
 
+    #setup Login Manager
+    login_manager = LoginManager()
+    login_manager.init_app(app)
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        
+        state = str(uuid.uuid4())
+        session['state']=state
+        url = f"https://github.com/login/oauth/authorize?client_id={os.getenv('OAUTH_CLIENT_ID')}&state={state}"
+
+        #If login is disabled for testing we can go straight to home page
+        if os.getenv('LOGIN_DISABLED')=='True':
+            return redirect("/")
```

Comment:
This doesn't do what you think it does. It gets your integration test to pass but for the wrong reasons.

Try setting LOGIN_DISABLED=True in your .env file and start up your app. I expect you'll see an infinite loop as your browser keeps trying to load the homepage and keeps getting a "redirect" to the same page.

What you want is to set `self.LOGIN_DISABLED = os.getenv('LOGIN_DISABLED')=='True'` inside your Config class in `flask_config.py`. This config setting will get picked up by the flask login library and when True, the "login_required" decorators will no longer do anything -  so you load the index page without ever calling this `unauthenticated` function. 

And then you should delete these two lines.